x-pack/solutions/search/test/api_integration/apis/search_playground/playground_utility_apis.ts

```diff
@@ -0,0 +1,291 @@
+          }),
+        });
+      });
+
+      it('should return search results for a test query', async () => {
+        const { body } = await supertestWithoutAuth
+          .post('/internal/search_playground/query_test')
+          .set('kbn-xsrf', 'xxx')
+          .auth(USERS.ALL.username, USERS.ALL.password)
+          .send({
+            elasticsearch_query: JSON.stringify({
+              retriever: {
+                standard: { query: { multi_match: { query: '{query}', fields: ['description'] } } },
+              },
+            }),
+            indices: ['search-national-parks'],
+            query: 'Banff',
+            chat_context: {
+              source_fields: '{"search-national-parks":["description"]}',
+              doc_size: 1,
+            },
+          })
+          .expect(200);
+
+        expect(body).toBeDefined();
+        expect(body.searchResponse).toBeDefined();
+      });
+
+      it('should return paginated search results for a search query', async () => {
+        const { body } = await supertestWithoutAuth
+          .post('/internal/search_playground/search')
+          .set('kbn-xsrf', 'xxx')
+          .auth(USERS.ALL.username, USERS.ALL.password)
+          .send({
+            elasticsearch_query: JSON.stringify({
+              retriever: {
+                standard: { query: { multi_match: { query: '{query}', fields: ['description'] } } },
+              },
+            }),
+            indices: ['search-national-parks'],
+            search_query: 'Banff',
+            size: 10,
+            from: 0,
+          })
+          .expect(200);
+
+        expect(body).toBeDefined();
+        expect(body.results).toBeDefined();
+        expect(body.executionTime).toBeDefined();
+        expect(body.pagination).toBeDefined();
+      });
+
+      it('should return the source fields for each index', async () => {
+        const { body } = await supertestWithoutAuth
+          .post('/internal/search_playground/query_source_fields')
+          .set('kbn-xsrf', 'xxx')
+          .auth(USERS.ALL.username, USERS.ALL.password)
+          .send({ indices: ['search-books', 'search-national-parks'] })
+          .expect(200);
+
+        expect(body).toBeDefined();
+        expect(body['search-books']).toBeDefined();
+        expect(body['search-national-parks']).toBeDefined();
+        expect(body['search-books'].source_fields).toEqual(
+          expect.arrayContaining(['author', 'name'])
+        );
+        expect(body['search-national-parks'].source_fields).toEqual(
+          expect.arrayContaining(['acres', 'date_established', 'description'])
+        );
+      });
+    });
+
+    describe('viewer', function () {
+      it('should NOT allow listing all available indices (returns 403)', async () => {
+        await supertestWithoutAuth
+          .get('/internal/search_playground/indices')
+          .set('kbn-xsrf', 'xxx')
+          .auth(USERS.READ.username, USERS.READ.password)
+          .expect(403);
```

Comment:
the `viewer` test in serverless returns `200`, we may need to update the `USERS.READ` roles to properly test these endpoints as the stack and serverless `viewer` tests should be functionally the same.